Add option to find peaks using ratio variance/ mean in FindSXPeaksConvolve

RichardWaiteSTFC opened the pull request:
### Description of work

Add option to find peaks using ratio variance/ mean in FindSXPeaksConvolve - this is a peak finding criterion used in DIALS, Winter (2018) 
https://doi.org/10.1107/S2059798317017235 

Here it is performed by doing 2 convolutions to evaluate `E[y]` and `E[y^2]` after scaling to ensure the data are in raw counts.

This work was started because it was noticed while benchmarking peak finding algorithms as part of testing #37123 that `FindSXPeaksConvolve` using ratio of I/sigma, was picking up lots of spurious noisy peaks in bank 1 of SXD  (higher gain as upgraded detector bank recently) unless quite a high value of I/sigma chosen.

Here is a comparison of the two methods for an SXD run with lower stats

![image](https://github.com/mantidproject/mantid/assets/55979119/1b5349d4-5f4d-42bb-9b84-30453a6a75e7)

It can be seen that the new method (variance/mean) find ~15-20% more peaks that can be indexed with 90% accuracy with a hkl tolerance of 0.1 (note detector positions were calibrated prior to this benchmark). it also out performs the older `FindSXPeaks` algorithm with `AllPEaks` strategy (see comments in #37123).

*There is no associated issue.*

### To test:

To-Do

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

RichardWaiteSTFC committed:
Add variance over mean ratio to FindSXPEaksConvolve

For a constant background the Poisson distributed counts should have Variance = mean (i.e. ratio = 1) - in the region of a Bragg peak the variance/mean > 1.
This idea is taken from DIALS software package for x-ray diffraction as described Winter (2018) https://doi.org/10.1107/S2059798317017235 - generalised to 3D TOF Laue

RichardWaiteSTFC committed:
Implement usign 1D convolutions (with uniform_filter) for performance

RichardWaiteSTFC committed:
Add argument doc strings and refactor to reduce size of main exec

RichardWaiteSTFC committed:
Add unit test

RichardWaiteSTFC committed:
Use FindSXPeaksConvolve as default in sxd and update system test

RichardWaiteSTFC committed:
Update documentation

RichardWaiteSTFC committed:
Add release notes

RichardWaiteSTFC committed:
Fix typo in docs

warunawickramasingha commented on Framework/PythonInterface/plugins/algorithms/FindSXPeaksConvolve.py:
```diff
@@ -109,16 +110,36 @@ def PyInit(self):
+            doc="Threshold value for variance/mean used to identify peaks.",
+        )
+        use_variance_over_mean = EnabledWhenProperty("PeakFindingStrategy", PropertyCriterion.IsNotDefault)
+        self.setPropertySettings("ThresholdVarianceOverMean", use_variance_over_mean)
```
From a user perspective, isn't it better to keep `ThresholdIoverSigma` and  `ThresholdVarianceOverMean` variables closer and next to each other that becomes active/inactive based on the selection of `PeakFindingStratergy` after it?

![image](https://github.com/mantidproject/mantid/assets/142818102/250305f6-8d49-4c28-a4ee-9b84e9a86b94)

warunawickramasingha replied:
yeah understood, I guess users won't bother much since there is only a small number of params to fill.

warunawickramasingha commented on docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst:
```diff
@@ -37,7 +37,7 @@ Here is an example using the ``SXD`` class that finds peaks and then removes dup
 
   # find peaks using SXD static method - determines peak threshold from
```
I guess this comment needs to be updated to represent Variance over mean

RichardWaiteSTFC committed:
Remove unnecessary reference file to avoid flaky tests

Test passes on windows not on jenkins linux, asserts test the functionality sufficiently

RichardWaiteSTFC committed:
Update technique docs as per PR review

warunawickramasingha reviewed (approved):
Thanks for updating the PR! The changes are working perfectly and happy to approve!
